apps/src/lib/kits/maker/boards/microBit/ExternalLed.js

```diff
@@ -12,11 +12,26 @@ export default class ExternalLed {
   }
 
   off() {
+    clearInterval(this.intervalID);
     this.board.setDigitalOutput(this.pin, 0);
     this.isOn = false;
   }
 
   toggle() {
     return this.isOn ? this.off() : this.on();
   }
+
+  blink(delay) {
+    this.off();
+    this.intervalID = setInterval(this.toggle_.bind(this), delay);
+  }
+
+  toggle_() {
+    return this.isOn ? this.off_() : this.on();
+  }
+
+  off_() {
```

Comment:
Minor, but I'm thinking if there are any alternate names we could use for the internal versions of toggle and off so it's more clear what these are doing and what the differences are...maybe `off_` could be something like `clearOutput`? Then your `off()` function would look like

```
clearInterval(intervalID);
this.clearOutput();
```

And if `toggle_` is only used within `blink`, maybe it's worth just moving that code into `blink` directly, so there's only one `toggle` function?

Definitely not ship-blocking, but just wanted to call this out cause it took me a bit to wrap my head around the differences between the functions.

Reply:
This is a great idea! 

And I'm actually going to update the `on` function to stop blinking before turning Led on (which is what happens with the Circuit Playground). So I'l go ahead and use `setDigitalOutputOn` and `setDigitalOutputOff` for the functions. Thanks!